api/v1/sriovnetworknodestate_types.go

```diff
@@ -83,6 +84,7 @@ type VirtualFunction struct {
 	Mtu        int    `json:"mtu,omitempty"`
 	VfID       int    `json:"vfID"`
 	VdpaType   string `json:"vdpaType,omitempty"`
+	GUID       string `json:"guid,omitempty"`
```

Comment:
generally need to deal with 3 cases here:

device is IB: GUID is present
device is eth but is not RDMA capable: GUID is empty/nil
device is eth and is RDMA capable: GUID is present, GUID represents the device "node guid"